### PR TITLE
WSTEAM1-130 Bylines TopicUrl

### PR DESCRIPTION
**Pull request by mfonofm**
Resolves #WSTEAM-1 130

**Overall change:**
Adds functionality in Optimo to link to an authors topic page

**Code changes:**

- Add function to get the `topicUrl` (as `authorTopicUrl`)
- Add `topicId` and `topicUrl` fields to the fixture data
- Add unit test

---

- [x] I have assigned myself to this PR and the corresponding issues
- [ ] I have added the `cross-team` label to this PR if it requires visibility across World Service teams
- [ ] I have assigned this PR to the Simorgh project
- [x] (BBC contributors only) This PR follows the [repository use guidelines](https://github.com/bbc/simorgh-infrastructure/blob/latest/documentation/repository-guidelines.md)

**Testing:**

- [ ] Automated (jest and/or cypress) tests added (for new features) or updated (for existing features)
- [ ] If necessary, I have run the local E2E non-smoke tests relevant to my changes (`CYPRESS_APP_ENV=local CYPRESS_SMOKE=false yarn test:e2e:interactive`)
- [ ] This PR requires manual testing

### CODE REVIEW

**Review by andrewscfc — COMMENTED** [review]
We should be adding a test for this change I would imagine?

File: src/app/pages/ArticlePage/Byline/fixture/index.js [hunk 1]
```diff
@@ -2,6 +2,8 @@ export const bylineWithNoRole = [
   {
     type: 'contributor',
     model: {
+      topicId: '',
+      topicUrl: '/news/topics/c8qx38nq177t',
```
Comment (amoore108) [inline src/app/pages/ArticlePage/Byline/fixture/index.js:6]: Is this where the `topicId` and `topicUrl` values will sit within the block? Do we have any data from Test Optimo or a set of example data of how it will look? Just imagined this to be more nested than sitting at the top level.

Comment (mfonofm) [reply src/app/pages/ArticlePage/Byline/fixture/index.js:6]: Yep, Jim shared this with me: https://jira.dev.bbc.co.uk/secure/attachment/1106690/defaultAfterAres.json
It looks like it's going to sit within the top level of the `contributor` block.

Comment (amoore108) [reply src/app/pages/ArticlePage/Byline/fixture/index.js:6]: Ah nice! Thanks for linking that 👍🏼

**Comment by amoore108** [comment]
Should this still be checking the `twitterLink`, should be `authorTopicUrl` I think?
https://github.com/bbc/simorgh/blob/86b6c1e14013dd8252617ceb545285a06e592550/src/app/pages/ArticlePage/Byline/index.tsx#L127

**Review by amoore108 — APPROVED** [review]

File: src/app/pages/ArticlePage/Byline/index.tsx [hunk 2]
```diff
@@ -151,6 +154,7 @@ const Byline = ({ blocks, children }: PropsWithChildren<Props>) => {
               <strong
                 css={[
                   BylineCss.author,
+                  BylineCss.authorLink,
                   getSansBold(service),
                   getBodyCopy(script),
```
Comment (andrewscfc) [inline src/app/pages/ArticlePage/Byline/index.tsx:159]: Should we getting the typography from the theme now? Can do this in another PR though

**Review by andrewscfc — APPROVED** [review]
looks great, one observation not directly related to this PR

File: src/app/pages/ArticlePage/Byline/index.tsx (hunk 2 above)
Comment (mfonofm) [reply src/app/pages/ArticlePage/Byline/index.tsx:159]: Yep, we could use the theme now. I'll write up a seperate ticket for this.
